website/programs.py

```diff
@@ -223,25 +223,30 @@ def share_unshare_program(self, user, program_id, loop_index):
         ):
             self.db.set_favourite_program(user["username"], None)
 
-        if bool(program["public"]) is True:
+        if program.get("public"):
             public = 0
         else:
             public = 1
         program = self.db.set_program_public_by_id(program_id, public)
         achievement = self.achievements.add_single_achievement(user["username"], "sharing_is_caring")
         if achievement:
-            utils.add_pending_achievement({"achievement": achievement, "redirect": "/programs"})
+            utils.add_pending_achievement({"achievement": achievement})
 
         keyword_lang = g.keyword_lang
         adventure_names = hedy_content.Adventures(g.lang).get_adventure_names(keyword_lang)
         program["date"] = utils.delta_timestamp(program["date"])
         program["preview_code"] = "\n".join(program["code"].split("\n")[:4])
         program["number_lines"] = program["code"].count('\n') + 1
-        return jinja_partials.render_partial('htmx-program-buttons.html',
+        if second_teachers_programs is True:
+            result = True
+        else:
+            result = False
```

Comment:
```suggestion
```

Reply:
@hasan-sh if I remove this, the dropdown dissapears after toggeling private/public

Reply:
yeah i see the issue.. cuz the second_teachers_programs is given as a string, and need to be checked for indeed. However, your if statement won't solve it since second_teachers_programs is never True. I'd do the following for now:

```second_teachers_programs = second_teachers_programs == 'True'```